### PR TITLE
mzcompose: Only pass extra args as -e, don't overwrite environment

### PR DESCRIPTION
**Pull request by def-**
Reported by Parker in https://materializeinc.slack.com/archives/CM7ATT65S/p1732031036476939?thread_ts=1732029052.219219&cid=CM7ATT65S

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Review by ParkMyCar — APPROVED** [review]

File: misc/python/materialize/mzcompose/composition.py [hunk 1]
```diff
@@ -788,7 +788,6 @@ def run(
             capture_and_print=capture_and_print,
             stdin=stdin,
             check=check,
-            environment=env_extra,
```
Comment (antiguru) [inline misc/python/materialize/mzcompose/composition.py:791]: That alone won't work, it needs this change above:
```
            *(f"-e{k}={v}" for k, v in env_extra.items()),
```

But that will produce very long log messages :/

Comment (def-) [reply misc/python/materialize/mzcompose/composition.py:791]: Oh, right. We can filter them out when printing, or pass in the original environment too. I'll do the latter for now.

Comment (ParkMyCar) [inline misc/python/materialize/mzcompose/composition.py:791]: another alternative would be something like:
```
environment={**os.environ, **env_extra},
```
this should ensure we retain whatever environment variables are already set

Comment (def-) [reply misc/python/materialize/mzcompose/composition.py:791]: Yes, that's pretty much what I did. It still works for me on macOS, so should be good

**Review by antiguru — APPROVED** [review]

**Pull request merged**